### PR TITLE
feat: allows to define a prefix for collections / tables

### PR DESCRIPTION
**Pull request by ytvnr**
Fixes https://github.com/gravitee-io/issues/issues/4715

### CODE REVIEW

File: src/main/resources/scripts/create-index.js [hunk 1]
```diff
@@ -1,115 +1,117 @@
...
+db.getCollection(`${prefix}genericnotificationconfigs`).createIndex( {"referenceId":1, "referenceType":1});
+db.getCollection(`${prefix}genericnotificationconfigs`).reIndex();
 
 // "alert triggers" collection
-db.alert_triggers.dropIndexes();
-db.alert_triggers.createIndex( { "referenceType": 1, "referenceId": 1 } );
-db.alert_triggers.reIndex();
+db.getCollection(`${prefix}alter_triggers`).dropIndexes();
+db.getCollection(`${prefix}alter_triggers`).createIndex( { "referenceType": 1, "referenceId": 1 } );
+db.getCollection(`${prefix}alter_triggers`).reIndex();
```
Comment (phiz71) [inline src/main/resources/scripts/create-index.js:106]:
```suggestion
db.getCollection(`${prefix}alert_triggers`).dropIndexes();
db.getCollection(`${prefix}alert_triggers`).createIndex( { "referenceType": 1, "referenceId": 1 } );
db.getCollection(`${prefix}alert_triggers`).reIndex();
```

File: src/main/java/io/gravitee/repository/mongodb/ratelimit/MongoRateLimitRepository.java [hunk 2]
```diff
@@ -48,6 +49,9 @@
     @Qualifier("rateLimitMongoTemplate")
     private ReactiveMongoOperations mongoOperations;
 
+    @Value("${management.mongodb.prefix:}")
```
Comment (phiz71) [inline src/main/java/io/gravitee/repository/mongodb/ratelimit/MongoRateLimitRepository.java:52]: What is the default value in that case: null or empty string ?

**Review by phiz71 — CHANGES_REQUESTED** [review]

File: src/main/java/io/gravitee/repository/mongodb/ratelimit/MongoRateLimitRepository.java (hunk 2 above)
Comment (phiz71) [reply src/main/java/io/gravitee/repository/mongodb/ratelimit/MongoRateLimitRepository.java:52]: It is an empty string.

File: src/main/resources/scripts/create-index.js (hunk 1 above)
Comment (ytvnr) [reply src/main/resources/scripts/create-index.js]: done

**Review by phiz71 — APPROVED** [review]

**Pull request merged**